### ISSUE TITLE
It seems like the website crashes sometime

### ISSUE THREAD

**Issue by OAHC2022** [awesome-schedule/Awesome-SchedulAR#10]
**The website at one point takes too long to load:**
<img width="1203" alt="issue_5_loading" src="https://user-images.githubusercontent.com/43483555/55832627-f2420600-5ae3-11e9-8139-e909d7abff04.png">

This problem is reflected by other user as well:
<img width="480" alt="issue_5_loading" src="https://user-images.githubusercontent.com/43483555/55832631-f5d58d00-5ae3-11e9-8648-9fbec04045af.png">

All of these happened around noon on 04/09/2019. Can't identify the cause. I speculate that it is the problem of the code because the website works fine for the most circumstances.

**Comment by hanzhi713** [comment]
First I would like to point out that the loading speed may be quite slow when the traffic is high, due to the fact that the cross-origin proxy may throttle. If Lou's list can enable cross-origin resource sharing, that will be great. 
 
Second, I fixed a small issue at 8a3a428226016362325d511473905354d539259b yesterday. Our program may get stuck in an infinite loop when there're more than 11 distinct classes selected because the color function is unable to probe any available color. Now I incorporated separate chaining to temporarily solve that issue. I'll further refine the automatic color selection algorithm when I have time. (Too much homework not done this week ahhhhh :frowning:). Users will be able to choose colors themselves in the future.
